jbpm-flow/src/main/java/org/jbpm/workflow/instance/node/SubProcessNodeInstance.java

```diff
@@ -201,6 +204,9 @@ public void internalTrigger(final NodeInstance from, String type) {
                 // remove foreach input variable to avoid problems when running in variable strict mode
                 parameters.remove(getSubProcessNode().getMetaData("MICollectionInput"));
             }
+            if (getVariable(ENTRY_POINT_VAR_NAME) != null) {
```

Comment:
This seems to happen at the end, so seems to override any value provided in mapping?  Should we only do this if this param isn't set yet?  Also would suggest just using getVariable(..) once and reus second time for efficiency.